src/simplewallet/simplewallet.cpp

```diff
@@ -6520,6 +6521,10 @@ void simple_wallet::check_for_inactivity_lock(bool user)
     }
     while (1)
     {
+      if (loop_counter > 3)
```

Comment:
Is there a reason why it can't be realized via:
`while (loop_counter++ <= 3)` ?

If there is, then it's safer to write:
`if (loop_counter++ > 3)`
and remove line 6536

Ideally, the `3` should be declared as a constant, so finally:

```C++
const int MAX_UI_LOOPS = 3;
while (loop_counter++ <= MAX_UI_LOOPS)
{
 /*... */
}
```